server/aesdsocket.c

```diff
@@ -0,0 +1,239 @@
+    			{
+    				perror("error changing directory");
+    				exit (-1);
+    			} 
+    			open("/dev/null", O_RDWR);
+    			dup(0);
+    			dup(0);
+    			printf("Daemon created successfully\n");
+    			//exit (0);
+    		}
+            else {
+                exit (0);
+            }
+        }
+    }
+
+    printf("Listening\n");
+    int sockls;
+    sockls = listen(sockfd, 5);
+    printf("Listening successful\n");
+
+    if (sockls == -1)
+    {
+        perror("Error Listening \n");
+        return -1;
+    }
+
+    printf("opening /var/tmp/aesdsocketdata\n");
+    file_fd = open("/var/tmp/aesdsocketdata",O_RDWR | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR | S_IRGRP |S_IWGRP | S_IROTH | S_IWOTH);
+    
+    if (file_fd == -1)
+    {
+        perror("error opening the file");
+    }
+    printf("file successfully opened\n");
+    int file_size = 0;
+    
+    char receive_buffer[BUFFER_SIZE];
+    
+    while(1)
+    {
+        client_len = sizeof(client_address);
+        clientfd = accept(sockfd, (struct sockaddr *)&client_address, &client_len);
+        if (clientfd == -1)
+        {
+            perror("error accepting\n");
+            return -1;
+        }
+        printf("Accepted connection from %s\n",inet_ntoa(client_address.sin_addr));
+        syslog(LOG_DEBUG,"Accepted connection from %s\n",inet_ntoa(client_address.sin_addr));  // log to syslog
+
+        bool packet_complete_status = false;
+        int buffer = BUFFER_SIZE;
+        int running_buffer_size = 0;
+        char *write_buffer = malloc(sizeof(char) * buffer);
+
+        while(!packet_complete_status)
+        {
+            receive_bytes = recv(clientfd, receive_buffer, BUFFER_SIZE, 0);
+            if (receive_bytes == 0)
+            {
+                packet_complete_status = true;
+            } 
+            for (int i = 0; i < receive_bytes; i++)
+            {
+                //printf("recieved byte is %d",receive_buffer[i]);
+                if (receive_buffer[i] == '\n')
+                {
+                    packet_complete_status = true;
+                }
+            }
+            if ((buffer - running_buffer_size) < receive_bytes)
+            {
+                buffer += receive_bytes;
+                write_buffer = (char *) realloc(write_buffer, sizeof(char) * buffer);
+            }
+            memcpy(write_buffer + running_buffer_size, receive_buffer, receive_bytes);
+            running_buffer_size += receive_bytes;
+        }
+        
+        if(packet_complete_status)
+        {
+            packet_complete_status = false;
+            // write to file
+            int write_bytes = write(file_fd, write_buffer, running_buffer_size);
+            if (write_bytes == -1)
+            {
+                perror("error writing to file");
+                return -1;
+            }
+            file_size += write_bytes;
+
+            lseek(file_fd, 0, SEEK_SET);
+
+            char *read_buffer = (char *) malloc(sizeof(char) * file_size);
```

Comment:
@TejaswiniLakshminarayan what happens when the total size of all packets sent are 1GB and you attempt to run this on qemu with 256 MB of RAM?  What assumption are you making about the total size of all packets sent here?